src/Kentico.Kontent.ModelGenerator.Core/Helpers/ElementTypeHelper.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using Kentico.Kontent.Management.Models.Types.Elements;
+using Kentico.Kontent.ModelGenerator.Core.Common;
+using Kentico.Kontent.ModelGenerator.Core.Configuration;
+
+namespace Kentico.Kontent.ModelGenerator.Core.Helpers
+{
+    public static class ElementTypeHelper
+    {
+        public static string GetElementType(CodeGeneratorOptions options, string elementType, ElementMetadataBase managementElement)
+        {
+            Validate(options, elementType, managementElement);
+
+            if (options.ContentManagementApi && elementType == "modular_content" && managementElement.Type == ElementMetadataType.Subpages)
```

Comment:
Basically get the answer to the question: why there is a difference between Delivery and Management SDK?

Reply:
The reaseno why we need to treat Delivery and Management api response differently because each api provides different type for subpages element. 
https://docs.kontent.ai/reference/delivery-api#section/Subpages-element
https://docs.kontent.ai/reference/management-api-v2#section/Subpages-element-in-type